actionpack/lib/action_dispatch/journey/gtg/simulator.rb

```diff
@@ -14,6 +14,8 @@ def initialize(memos)
       end
 
       class Simulator # :nodoc:
+        INITIAL_STATE = [0].freeze
+        MEMO_PATTERN = %r([/.?]|[^/.?]+).freeze
```

Comment:
Think Ruby already dedups Regexes at parse time, so this constant shouldn't be needed.

Reply:
Cool! Didn't know that.